exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java

```diff
@@ -277,6 +279,25 @@ public String deriveAlias(
       return SqlValidatorUtil.getAlias(node, ordinal);
     }
 
+    /**
+     * Checks that specified expression is not implicit column and
+     * adds is to a select list, ensuring that its alias does not
+     * clash with any existing expressions on the list.
+     */
+    @Override
+    protected void addToSelectList(
```

Comment:
Could you please add comment explaining when this method is used and how know when to add columns to Avro table and not for Dynamic?

Reply:
Added comment into Javadoc, which describes when and why this method is used.

As for Avro and Dynamic tables, the key point is the result of `RelDataType.isDynamicStruct()` method. 
For `AvroDrillTable` we use `ExtendableRelDataType` whose `isDynamicStruct()` method returns `false`, but for `DynamicDrillTable` we use `RelDataTypeDrillImpl`  whose `isDynamicStruct()` method returns `true`. 
In such way, Calcites `SqlValidatorImpl` determines whether columns should be added.